pkgs/development/compilers/mit-scheme/default.nix

```diff
@@ -29,6 +30,8 @@ stdenv.mkDerivation {
       sha256 = "0w5ib5vsidihb4hb6fma3sp596ykr8izagm57axvgd6lqzwicsjg";
     };
 
+  buildInputs = if enableX11 then [xlibsWrapper] else [];
```

Comment:
Could be `stdenv.lib.optionals enableX11 [ xlibsWrapper ]`

Reply:
Actually even `buildInputs = stdenv.lib.optional enableX11 xlibsWrapper;` (optional**s** -> optional)

Reply:
It'd be better to have `buildInputs` as a list so that additional build inputs may be added in an override. It also doesn't scale well, if this derivation would ever need another build input, it will have to switch back to `optionals`.